### PR TITLE
Improved error message for package manager

### PR DESCRIPTION
**Pull request by sdstolworthy**
## Problem

Directs users to use `npx amplify` if the CLI errors when not being run with a package manager.

**Issue number, if available:**

## Changes

**Corresponding docs PR, if applicable:**

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by 0618 — DISMISSED** [review]

File: packages/cli-core/src/package-manager-controller/package_manager_controller_factory.ts [hunk 1]
```diff
@@ -57,7 +57,9 @@ export class PackageManagerControllerFactory {
   private getPackageManagerName() {
     const userAgent = process.env.npm_config_user_agent;
     if (userAgent === undefined) {
-      throw new Error('npm_config_user_agent is undefined');
+      throw new Error(
```
Comment (edwardfoyle) [inline packages/cli-core/src/package-manager-controller/package_manager_controller_factory.ts:60]: We should use `AmplifyUserError` now that we have it

File: packages/cli-core/src/package-manager-controller/package_manager_controller_factory.ts [hunk 2]
```diff
@@ -57,7 +57,10 @@ export class PackageManagerControllerFactory {
   private getPackageManagerName() {
     const userAgent = process.env.npm_config_user_agent;
     if (userAgent === undefined) {
-      throw new Error('npm_config_user_agent is undefined');
+      throw new AmplifyUserError('NoPackageManagerError', {
+        resolution: `This is usually caused by attempting to run \`amplify\` directly. Try running \`npx amplify\``,
```
Comment (edwardfoyle) [inline packages/cli-core/src/package-manager-controller/package_manager_controller_factory.ts:61]: nit: could use `format.command('npx amplify')`

**Review by edwardfoyle — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

File: packages/cli-core/src/package-manager-controller/package_manager_controller_factory.ts (hunk 2 above)
Comment (edwardfoyle) [reply packages/cli-core/src/package-manager-controller/package_manager_controller_factory.ts]: We have `format.runner('npx').amplifyCommand('sandbox')` for example, but that requires the caller to have resolved which package manager is being used. Which is what we are trying to get from the `npm_config_user_agent` but it's not set, so I'm not sure how we can do any better in this case.

**Review by edwardfoyle — APPROVED** [review]

File: packages/cli-core/src/package-manager-controller/package_manager_controller_factory.ts (hunk 2 above)
Comment (sobolk) [reply packages/cli-core/src/package-manager-controller/package_manager_controller_factory.ts]: `Try running with package manager, for example \`npx amplify\`

**Pull request merged**